powermon/protocols/abstractprotocol.py

```diff
@@ -123,65 +123,16 @@ def get_command_with_command_string(self, command) -> CommandDefinition:
         log.info("No command_defn found for %s", command)
         return None
 
-    def get_responses(self, response) -> list:
-        """
-        Default implementation of split and trim
-        """
-        # Trim leading '(' + trailing CRC and \r of response, then split
-        return response[1:-3].split(None)
 
-    def check_response_valid(self, result: Result) -> Result:
+
+    def check_response_and_trim(self, response: str) -> str:
         """
         Simplest validity check, CRC checks should be added to individual protocols
         """
-        if result.raw_response is None:
-            result.is_valid = False
-            result.error = True
-            result.error_messages.append("failed validity check: response was empty")
+        if response is None:
+            raise ValueError("Response is None")
```

Comment:
I wonder if we should expand the exceptions we use - powermon.Errors.InvalidResponse etc, so we can deal with them 'more properly'

Reply:
Yeah we can include more errors when we get to the point of handling expected errors gracefully.